core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1261,4 +1265,13 @@ public void setNonChMaxWaypointDistance(int nonChMaxWaypointDistance) {
         this.nonChMaxWaypointDistance = nonChMaxWaypointDistance;
     }
 
+    public void setSpatialRuleLookup(SpatialRuleLookup spatialRuleLookup) {
```

Comment:
Makes sense :+1: I think it makes sense to start only with generic for now. We can always extend that.